contracts/tokenLocking/TokenLocking.sol

```diff
@@ -195,7 +195,14 @@ contract TokenLocking is TokenLockingStorage, DSMath, BasicMetaTransaction { //
     Lock storage userLock = userLocks[_token][_user];
     userLock.balance = sub(userLock.balance, _amount);
 
-    makeConditionalDeposit(_token, _amount, _recipient);
+    if (_recipient == address(0x0)) {
+      // If the burn fails, transfer to 0x0
+      try ERC20Extended(_token).burn(_amount) {} catch {
+        require(ERC20Extended(_token).transfer(address(0x0), _amount), "colony-token-locking-burn-failed");
```

Comment:
I'm not following -- you're saying someone could design a token contract which fails? We `require` transfers all the time, so I'm not sure why this is different. Would removing the special handling for `burn` solve the issue?

Reply:
>We require transfers all the time, so I'm not sure why this is different 

Just because this is on TokenLocking. Currently, this function can never revert, and I want it to keep that property (so long as the tokens can always end up somewhere appropriate, which they can, as we can always move them via internal bookkeeping) so that when we're doing higher-level work with extensions, we don't have to stop and think 'how is the flow affected if this call, which three calls down in the stack calls `transferStake` on TokenLocking, can never succeed'.

Reply:
Sure, but the only scenarios in which this can occur is when a colony is using a malicious token as their internal token, which I have a hard time seeing as occurring by accident. Further, it would only affect that colony, which ostensibly wouldn't be able to execute any token transfers at all. So this seems like the least of their worries.

Reply:
> which ostensibly wouldn't be able to execute any token transfers at all

A malicious token can behave normally and be made malicious at will at a judicious time, or even for a single judiciously chosen call. Or maybe there's no malice at all, and a token upgrade goes wrong, causing it to revert calls for a period of time.